k3s: documentation updates (#314395)
NixOS/nixpkgs: pkgs/applications/networking/cluster/k3s/README.md

# K3s

K3s is a simplified [Kubernetes](https://wiki.nixos.org/wiki/Kubernetes) version that bundles  Kubernetes cluster components into a few small binaries optimized for Edge and IoT devices.

## Usage

> * [Module Usage](docs/USAGE.md).

Reviewer: ```suggestion
* [Module Usage](docs/USAGE.md)
```

## Configuration Examples

* [Nvidia GPU Passthru](docs/examples/NVIDIA.md)
* [Storage Examples](docs/examples/STORAGE.md)

## Cluster Maintenance and Troubleshooting

> * [Cluster Upkeep](docs/CLUSTER_UPKEEP.md).

Reviewer: ```suggestion
* [Cluster Upkeep](docs/CLUSTER_UPKEEP.md)
```

# K3s

K3s is a simplified [Kubernetes](https://wiki.nixos.org/wiki/Kubernetes) version that bundles  Kubernetes cluster components into a few small binaries optimized for Edge and IoT devices.

## Usage

Starter usage examples can be found in [docs/USAGE.md](docs/USAGE.md). 

## Configuration Examples

* [ZFS Snapshot Support](docs/examples/ZFS.md)
* [Nvidia GPU Passthru](docs/examples/NVIDIA.md)
* [Storage Examples](docs/examples/STORAGE.md)

## Cluster Maintenance and Troubleshooting

For cluster maintenance and troubleshooting information, please check out [docs/CLUSTER_UPKEEP.md](docs/CLUSTER_UPKEEP.md).

## K3s Package Upkeep

* [Package Versioning Rationale](docs/VERSIONING.md)

> * [Package Maintenance Documentation][docs/PKG_UPKEEP.md]

Author: ```suggestion
* [Package Maintenance Documentation](docs/PKG_UPKEEP.md)
```

# K3s

> K3s is a simplified version of [Kubernetes](https://wiki.nixos.org/wiki/Kubernetes). It bundles all components for a kubernetes cluster into a few of small binaries.

Reviewer: ```suggestion
K3s is a simplified [Kubernetes](https://wiki.nixos.org/wiki/Kubernetes) version that bundles  Kubernetes cluster components into a few small binaries optimized for Edge and IoT devices.
```

## Usage

> Starter usage examples can be found in [docs/USAGE.md](docs/USAGE.md). 

Reviewer: ## [Usage examples](docs/USAGE.md)

Author: Fine with removing verbiage, but not a fan of inconsistent header usage. "Are they links or are they headers?" type thought. I am thinking just an array of links below each heading and at least be consistent with that in the README across all sections.

Reviewer: Sure, can do like that. Keep it short.

## Configuration Examples

> The following documents help deal with special use cases and real work examples for cluster setup.

Reviewer: ```suggestion
```

> * [ZFS Snapshot Support](docs/examples/ZFS.md)

Reviewer: Isn't ZFS part of storage?

Author: It was its own sub-section of the Wiki, quick copied to its own file. No qualms moving it into storage.

> * [Nvidia Support](docs/examples/NVIDIA.md)

Reviewer: ```suggestion
* [Nvidia GPUs Passthru](docs/examples/NVIDIA.md)
```

* [Storage Examples](docs/examples/STORAGE.md)

> ## Cluster Maintenance and Troubleshooting

Reviewer: ```suggestion
## [Cluster Maintenance and Troubleshooting](docs/CLUSTER_UPKEEP.md)
```

Author: Same thought as https://github.com/NixOS/nixpkgs/pull/314395#discussion_r1614290529

> For cluster maintenance and troubleshooting information, please check out [docs/CLUSTER_UPKEEP.md](docs/CLUSTER_UPKEEP.md).

Reviewer: ```suggestion
```

Author: Based on previous comments on comments as headers, thought would be then to just change this to something like: 

```
* [Cluster Upkeep](docs/CLUSTER_UPKEEP.md)
```

Reviewer: Feel free to organize things as you wish. I don't want to nitpick. I just wanted to convey to you the idea we should drop as many words as possible. To make things concise. Goal should be to make documentation shorter but meaningful (concise). As I explained in another comment. If we are able to do it, it is going to a nice one.

## K3s Package Upkeep

> Versioning and package upkeep documentation can be found in the following locations:

Reviewer: ```suggestion
```
